source/common/http/utility.cc

```diff
@@ -148,39 +149,18 @@ bool Utility::isWebSocketUpgradeRequest(const HeaderMap& headers) {
                     Http::Headers::get().UpgradeValues.WebSocket.c_str())));
 }
 
-Http2Settings Utility::parseHttp2Settings(const Json::Object& config) {
+Http2Settings Utility::parseHttp2Settings(const envoy::api::v2::Http2ProtocolOptions& config) {
   Http2Settings ret;
-
-  Json::ObjectSharedPtr http2_settings = config.getObject("http2_settings", true);
-  ret.hpack_table_size_ =
-      http2_settings->getInteger("hpack_table_size", Http::Http2Settings::DEFAULT_HPACK_TABLE_SIZE);
-  ret.max_concurrent_streams_ = http2_settings->getInteger(
-      "max_concurrent_streams", Http::Http2Settings::DEFAULT_MAX_CONCURRENT_STREAMS);
-  ret.initial_stream_window_size_ = http2_settings->getInteger(
-      "initial_stream_window_size", Http::Http2Settings::DEFAULT_INITIAL_STREAM_WINDOW_SIZE);
+  ret.hpack_table_size_ = MessageUtil::getUInt32(config, "hpack_table_size",
```

Comment:
Actually, I'm not sure it really required any macros. You could just have utility functions that look like `getOrReturnDefault(..., ...)` where you pass the value of a proto method call into the function and if certain things aren't set, etc. either return default, throw, etc.?